src/lightning/pytorch/loggers/utilities.py

```diff
@@ -69,6 +69,9 @@ def _log_hyperparams(trainer: "pl.Trainer") -> None:
         lightning_hparams = pl_module.hparams_initial
         inconsistent_keys = []
         for key in lightning_hparams.keys() & datamodule_hparams.keys():
+            if key.startswith("_"):
+                # Prevent merging LightningCLI's internal hparams and discourage of private param names
```

Comment:
can we throw a warning here when a private param is skipped instead of doing it silently?

Reply:
I was adding a warning, but then realized that if `LightningCLI` is used, then there would always be a warning. So now I changed the code so that only the `LightningCLI` internal params are skipped and removed from logging.

Reply:
Also only skipping `_class_path` because if `_instantiator` is skipped, then `test_lightning_cli_load_from_checkpoint_dependency_injection` fails. I don't understand why, but anyway `_class_path` is the only problem in #20182.

Reply:
great, thanks @mauvilsa